src/golang/lib/models/shared/execution.go

```diff
@@ -31,6 +31,18 @@ func (e ExecutionState) Terminated() bool {
 	return e.Status == FailedExecutionStatus || e.Status == SucceededExecutionStatus || e.Status == CanceledExecutionStatus
 }
 
+func (e *ExecutionState) HasBlockingFailure() bool {
+	return e.Status == FailedExecutionStatus && *e.FailureType != UserNonFatalFailure
+}
+
+func (e *ExecutionState) HasWarning() bool {
+	return e.Status == FailedExecutionStatus && !e.HasBlockingFailure()
+}
+
+func (e *ExecutionState) HasSystemError() bool {
+	return e.Status == FailedExecutionStatus && *e.FailureType == SystemFailure
+}
```

Comment:
wait why are there two ExecutionStates that mirror each other? If it's unavoidable can we please make this a big banner comment on both files?

Reply:
It's not specific to execution types, it's captured by https://linear.app/aqueducthq/issue/ENG-1934/move-utils-in-collections-to-repos-as-necessary that some `collections/shared` and `models/shared` are duplicated since callers are not fully migrated